README.md

```diff
@@ -1,24 +1,170 @@
-# README
+<a name="readme-top"></a>
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+## 📗 Table of Contents
 
-Things you may want to cover:
+- [Greeting App ](#greeting-app-)
+    - [Tech Stack ](#tech-stack-)
+    - [Key Features ](#key-features-)
+  - [💻 Getting Started ](#-getting-started-)
+    - [Prerequisites ](#prerequisites-)
+    - [Setup ](#setup-)
+    - [Installation ](#installation-)
+    - [database ](#database-)
+    - [Usage ](#usage-)
+  - [✍️ Authors ](#️-authors-)
+  - [🔭 Future Features ](#-future-features-)
+  - [🤝 Contributing ](#-contributing-)
+  - [💖 Show your support ](#-show-your-support-)
+  - [🙏 Acknowledgments ](#-acknowledgments-)
+  - [📝  License ](#--license-)
 
-* Ruby version
 
-* System dependencies
+# Greeting App <a name="about-project"></a>
 
-* Configuration
+**This web app combines React, Rails, and Webpack to provide users with a random greeting upon each visit. React handles the frontend, Rails manages the backend with a RESTful API, and Webpack optimizes asset bundling for a seamless and personalized user experience.**.<a name="built-with"></a>
 
-* Database creation
+### Tech Stack <a name="tech-stack"></a>
 
-* Database initialization
+<summary>FrontEnd</summary>
+  <ul>
+    <li>HTML</li>
+    <li>CSS</li>
+    <li>JavaScript</li>
+    <li>React</li>
+    <li>Redux</li>
+  </ul>
 
-* How to run the test suite
+<summary>BackEnd</summary>
+<ul>
+  <li>Ruby</li>
+</ul>
 
-* Services (job queues, cache servers, search engines, etc.)
+<summary>Frameworks</summary>
+<ul>
+  <li>Rails</li>
+</ul>
 
-* Deployment instructions
+<summary>Database</summary>
+<ul>
+  <li>PostgreSQL</li>
+</ul>
 
-* ...
+### Key Features <a name="key-features"></a>
+
+- **Generate random greeting**
+  
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+To get a local copy up and running, follow these steps.
+
+### Prerequisites <a name="prerequisites"></a>
+
+In order to run this project you need [Ruby](https://www.ruby-lang.org/en/), [Rails](https://rubyonrails.org/), [postgres](https://www.postgresql.org/) installed on your machine, and [react](https://react.dev/)
+
+
+### Setup <a name="setup"></a>
+
+Clone this repository to your desired folder:
+
+```sh
+  git clone https://github.com/yusufyusie/hello-rails-react.git
+```
+
+```sh
+  cd hello-rails-react
+```
+
+
+### Installation <a name="installation"></a>
+
+Install this project with:
+
+```sh
+  bundle install
+```
+
+Install node dependencies with:
+
+```sh
+npm install
+```
+
+Build project with:
+
+```sh
+npm run build
+```
```

Comment:
- [ ] Well! good job so far. but still need some changes. Currenty you added this command to build the project but when I run `npm run build` it give a error. I notic that in the package.json there want no such script. So I will reocmd you to fix this error and make sure your project work properly without any error.

![image](https://github.com/yusufyusie/hello-rails-react/assets/55840999/d9dfc1aa-e3e5-49d9-8da0-e2c38b7cbccd)